src/core/Ekubo.sol

```diff
@@ -72,13 +72,15 @@ library UnsafeEkuboCore {
             let ptr := mload(0x40)
 
             mstore(ptr, 0x00000000) // selector for `swap_611415377((address,address,bytes32),int128,bool,uint96,uint256)`
-            // TODO: Check that native ETH is handled properly (0x0 vs 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee)
-            mcopy(add(0x20, ptr), poolKey, 0x60)
+            let token0 := mload(poolKey)
+            token0 := mul(token0, iszero(eq(0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee, token0)))
+            mstore(add(0x20, ptr), token0)
+            mcopy(add(0x40, ptr), add(0x20, poolKey), 0x40)
```

Comment:
You should check this, but I think you may save on contract size if you
```
let firstArg := add(0x20, ptr)
mcopy(firstArg, poolKey, 0x60)
let token0 := mload(firstArg)
mstore(firstArg, mul(iszero(eq(0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee, token0)), token0))
```

Reply:
You are right
```
╭---------------------------------+------------------+-------------------+--------------------+---------------------╮
| Contract                        | Runtime Size (B) | Initcode Size (B) | Runtime Margin (B) | Initcode Margin (B) |
+===================================================================================================================+
| MainnetSettler   (Original)     | 21,803           | 23,361            | 2,773              | 25,791              |
| MainnetSettler   (   New  )     | 21,785           | 23,343            | 2,791              | 25,809              |
╰---------------------------------+------------------+-------------------+--------------------+---------------------╯
```